server/src/models/note.model.ts

```diff
@@ -8,6 +8,17 @@ interface NoteData {
   creationDate: number;
   lastUpdated: number;
 }
+interface EditableNoteField {
+  title?: string;
+  content?: string;
+  lastUpdated: number;
+}
+interface CreateNoteFields {
```

Comment:
A better way to write those type would be to make `CreateNoteFields` extend `EditableNoteFields`. Then you would only add the remaining fields in `CreateNoteFields`. 

```ts
interface CreateNoteFields extends Omit<EditableNoteFields, 'lastUpdated'> {
creationDate: number;
author: string;
}
```

Reply:
This is a not a blocking comment. We can always refactor the code later, I will approve the PR.

But it would nice if you could read about it and change it yourself.